### ISSUE TITLE
Error message for labeled-statement inside constexpr function body

### ISSUE THREAD

**Issue by ebdd6ebc-1612-4e9b-aee1-f7d2b6361d5f** [llvm/llvm-project#24621]
|  |  |
| --- | --- |
| Bugzilla Link | [24247](https://llvm.org/bz24247) |
| Version | trunk |
| OS | All |
| CC | @zygoloid |

## Extended Description 
Consider the following program (`prog.cc`):
```cpp
  constexpr void f() { x: ; }
  int main() {}
```
It is well-formed. Observe that [dcl.constexpr]/4 (http://eel.is/c++draft/dcl.dcl#dcl.constexpr-3) does forbid `goto` statements, but not labeled-statements.

Compile it with the following command-line:
```
  clang++ prog.cc -std=c++14 -pedantic-errors
```
The following error message is given:
```
  prog.cc:1:22: error: statement not allowed in constexpr function
  constexpr void f() { x: ; }
                     ^
  1 error generated.
```
Expected behaviour is to get no error messages since the program is well-formed.

I tired it with gcc also, and it gives no error message.

**Comment by shafik** [comment]
This is now a c++23 feature: https://godbolt.org/z/64GfjE77E

So closing.
